### DISCUSSION TITLE
PocketBase NextJS cache / revalidate

### DISCUSSION THREAD

**Discussion by LL33ch** [pocketbase/pocketbase#4993] [Q&A]
Hi, I'm using Nextjs 14, I wrote a route handler to get records from pb, and I need to cache some records so that I don't get actual data every time, or get them [on-demand](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#on-demand-revalidation).

I tried to use revalidate in the route handler, but it did not lead to the desired result, every time I access the route handler, it gives me the actual data, not cached.
```typescript
import { initPocketbaseServer } from '@/lib/initPocketbaseServer';

export async function GET() {
  const pb = await initPocketbaseServer();
  const res = await pb.collection('store_settings').getOne('97f3wcnhj37tek4', {
    next: { revalidate: 3600 },
  });

  return Response.json(res);
}
``` 

I tried to use revalidate in component and that also came to nothing:
```typescript
const response = await fetch(`http://${process.env.VERCEL_URL}/api/store_info`, {
    next: { revalidate: 3600 },
  });
  if (!response.ok) {
    throw new Error('Failed to upload store information');
  }
  const storeInfo: StoreInfo = await response.json();
```

**Comment by ganigeorgiev** [comment]
This is not really PocketBase specific and Next.js questions are better addressed at its related support channels.

~In any case there is no such fetch option as `next: { revalidate: 3600 }`.~ _I stand corrected as this seems to be part of the [Next.js Time based Revalidation](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#time-based-revalidation) but still this is not really PocketBase related so please refer to the below opting-out link_.

Try with `cache: 'no-store'` or `revalidate: 0`. For other options you can check the ["Opting out of Data Caching"](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#opting-out-of-data-caching) Next.js docs.

**Reply by LL33ch** [reply]
No, I need to cache the data, preferably so that I can update it [on-demand](https://nextjs.org/docs/app/building-your-application/caching#on-demand-revalidation). Does such a possibility exist or does this not apply to PocketBase?
